src/components/Timeline/Playbook/TextFilter.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component } from 'react'
+import { TextField, FlatButton } from 'material-ui'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+import * as filterActions from 'actions/filterActions'
+
+class TextFilter extends Component {
+  static propTypes = {
+    history: PropTypes.object.isRequired,
+    filters: PropTypes.object
+  }
+
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      value: ''
+    }
+  }
+
+  render () {
+    return (
+      <div>
+        <br />
+        <TextField
+          hintText='Search'
+          id='search-field-value'
+          value={this.state.value}
+          onChange={(event) => this.handleChange(event)}
+                />
+        <FlatButton
+          label='Search'
+          primary
+          onClick={() => this.submitSearch()}
+                />
+      </div>
+    )
+  }
+
+  handleChange (event) {
+    this.setState({
+      value: event.target.value
+    })
+  };
+
+  submitSearch () {
+    const { filters, dispatch, history } = this.props
+    const newFilter = Object.assign({}, filters, { DataSearch: this.state.value })
+    dispatch(filterActions.synchronizeFilters(newFilter, null, null, history))
+  }
+}
+
+export const mapStateToProps = (state, ownProps) => {
+  return {
+    ...ownProps,
+    dispatch: state.dispatch,
```

Comment:
Dispatch is added automatically, it doesn't have to be added manually in mapStateToProps